### PR TITLE
UI improvement to show gas amounts with in a room

### PR DESCRIPTION
**Pull request by mattpott**
Added ui improvement to room gasses so that they are now rounded to a set decimal place instead of showing the full float

Added dictionary to track if a gas is increasing or decreasing

### CODE REVIEW

File: Assets/Scripts/Models/Room.cs [hunk 1]
```diff
@@ -21,13 +21,15 @@ public class Room : IXmlSerializable
 {
     // Dictionary with the amount of gas in room stored in preasure(in atm) multiplyed by number of tiles
     private Dictionary<string, float> atmosphericGasses; 
+    public Dictionary<string, string> deltaGas;
```
Comment (Grenkin1988) [inline Assets/Scripts/Models/Room.cs]: No public fields.

File: Assets/Scripts/UI/MouseOverRoomDetails.cs [hunk 2]
```diff
@@ -55,9 +56,8 @@ void Update()
 
         foreach (string g in t.Room.GetGasNames())
         {
-            s += g + ": " + t.Room.GetGasPressure(g) + " (" + (t.Room.GetGasPercentage(g) * 100) + "%) ";
+            s += g + ": " + "("+ t.Room.deltaGas[g] + ") " + MathUtilities.RoundFloat(t.Room.GetGasPressure(g), 3) + " atm" + " (" + MathUtilities.RoundFloat(t.Room.GetGasFraction(g) * 100, 2) + "%) \n";
```
Comment (Grenkin1988) [inline Assets/Scripts/UI/MouseOverRoomDetails.cs]: g is not good name for variable. Please split line in two

File: Assets/Scripts/Utilities/MathUtilities.cs [hunk 3]
```diff
@@ -60,4 +61,9 @@ public static bool IsZero(this float value)
     {
         return Math.Abs(value) < double.Epsilon;
     }
+    public static float RoundFloat(this float value, int decimalplace)
```
Comment (Grenkin1988) [inline Assets/Scripts/Utilities/MathUtilities.cs]: No UnityEngine here

File: Assets/Scripts/UI/MouseOverRoomDetails.cs [hunk 4]
```diff
@@ -55,9 +56,8 @@ void Update()
 
         foreach (string g in t.Room.GetGasNames())
```
Comment (Grenkin1988) [inline Assets/Scripts/UI/MouseOverRoomDetails.cs]: Change name of g to gasName

File: Assets/Scripts/UI/MouseOverRoomDetails.cs [hunk 5]
```diff
@@ -55,9 +56,8 @@ void Update()
...
         {
-            s += g + ": " + t.Room.GetGasPressure(g) + " (" + (t.Room.GetGasPercentage(g) * 100) + "%) ";
+            s += g + ": " + "("+ t.Room.ChangedGases(g) + ") " + MathUtilities.RoundFloat(t.Room.GetGasPressure(g), 3) + " atm" + " (" + MathUtilities.RoundFloat(t.Room.GetGasFraction(g) * 100, 2) + "%) \n";
```
Comment (Grenkin1988) [inline Assets/Scripts/UI/MouseOverRoomDetails.cs]: Split in two line. If would be easier to read

Comment (Grenkin1988) [inline Assets/Scripts/UI/MouseOverRoomDetails.cs]: If woul use string.Format("format", params) here

File: Assets/Scripts/UI/MouseOverRoomDetails.cs [hunk 6]
```diff
@@ -54,9 +54,13 @@ void Update()
...
 
-        foreach (string g in t.Room.GetGasNames())
+        foreach (string gasName in t.Room.GetGasNames())
         {
-            s += g + ": " + "("+ t.Room.ChangedGases(g) + ") " + MathUtilities.RoundFloat(t.Room.GetGasPressure(g), 3) + " atm" + " (" + MathUtilities.RoundFloat(t.Room.GetGasFraction(g) * 100, 2) + "%) \n";
+            s += gasName + ": " + "(" + t.Room.ChangedGases(gasName) + ") ";
+
+            //s += MathUtilities.RoundFloat(t.Room.GetGasPressure(gasName), 3) + " atm";
+            s += string.Format("{0:0.000}", t.Room.GetGasPressure(gasName)) + " atm";
+            s += " (" + string.Format("{0:0.0}", t.Room.GetGasFraction(gasName)*100) + "%)\n";
```
Comment (Grenkin1988) [inline Assets/Scripts/UI/MouseOverRoomDetails.cs]: string.Format("{0}:  ({1}) {3} atm ({4}%", gasName, t.Room...) If mean line this :-)

File: Assets/Scripts/UI/MouseOverRoomDetails.cs [hunk 7]
```diff
@@ -53,11 +54,12 @@ void Update()
 
         string s = "";
 
-        foreach (string g in t.Room.GetGasNames())
+        foreach (string gasName in t.Room.GetGasNames())
         {
-            s += g + ": " + t.Room.GetGasPressure(g) + " (" + (t.Room.GetGasPercentage(g) * 100) + "%) ";
+            s += gasName + ": " + "(" + t.Room.ChangedGases(gasName) + ") ";
+            s += string.Format("{0:0.000}", t.Room.GetGasPressure(gasName)) + " atm";
+            s += " (" + string.Format("{0:0.0}", t.Room.GetGasFraction(gasName)*100) + "%)\n";
```
Comment (Grenkin1988) [inline Assets/Scripts/UI/MouseOverRoomDetails.cs]: string.Format("{0}: ({1}) {3} atm ({4}%", gasName, t.Room...) If mean line this :-)

**Comment by GamerInspired** [comment]
Any chance of an image of the UI improvement ?

File: Assets/Scripts/UI/MouseOverRoomDetails.cs (hunk 7 above)
Comment (mattpott) [inline Assets/Scripts/UI/MouseOverRoomDetails.cs]: :p sorry I had not worked in c# before

**Comment by mattpott** [comment]
Old
room_gas_ui_old

New
room_gas_ui

**Comment by vogonistic** [comment]
Beautiful!

**Comment by koosemose** [comment]
Very nice.

**Comment by GamerInspired** [comment]
Good job :smiley:

File: Assets/StreamingAssets/Settings/Settings.xml [hunk 8]
```diff
@@ -2,8 +2,8 @@
   <worldWidth>101</worldWidth>
   <worldHeight>101</worldHeight>
   <localization>en_US</localization>
-  <DialogBoxSettings_langToggle>True</DialogBoxSettings_langToggle>
-  <DialogBoxSettings_fpsToggle>True</DialogBoxSettings_fpsToggle>
+  <DialogBoxSettings_langToggle>False</DialogBoxSettings_langToggle>
+  <DialogBoxSettings_fpsToggle>False</DialogBoxSettings_fpsToggle>
```
Comment (vogonistic) [inline Assets/StreamingAssets/Settings/Settings.xml]: These changes shouldn't have been part of this commit.

**Comment by Mizipzor** [comment]
Tested this. But my master branch looks the same with or without merging this. Was ti merged as part of something else? Oh well, it doesnt appear to be breaking anything. Merged. :)

**Pull request merged**